screen/OrderData.xml.ftl

```diff
@@ -0,0 +1,172 @@
+              <#if !payment.isEmpty()>
+                  <#assign paymentMethod = ec.entity.find("mantle.account.method.PaymentMethod").condition("paymentMethodId",payment[0].paymentMethodId!).one()>
+                  <#assign paymentDescription = ec.entity.find("moqui.basic.Enumeration").condition("enumId",paymentMethod.paymentMethodTypeEnumId!).useCache(true).one()>
+                  <PaymentMethod>
+                     <![CDATA[${paymentDescription.description!}]]>
+                  </PaymentMethod>
+              </#if>
+              <CurrencyCode>${order.currencyUom!}</CurrencyCode>
+              <OrderTotal>${orderHeaderAndPart.partTotal!}</OrderTotal>
+              <#--
+              <TaxAmount>XX</TaxAmount>
+              -->
+              <ShippingAmount>${part.shippingCost!}</ShippingAmount>
+              <CustomerNotes>
+                 <![CDATA[${part.shippingInstructions!}]]>
+              </CustomerNotes>
+              <InternalNotes>
+                 <![CDATA[${part.shippingInstructions!}]]>
+              </InternalNotes>
+              <Gift>${orderHeaderAndPart.isGift!}</Gift>
+              <GiftMessage>${orderHeaderAndPart.giftMessage!}</GiftMessage>
+              <Customer>
+                 <CustomerCode>
+                    <![CDATA[${order.customer_details.email!}]]>
+                 </CustomerCode>
+                 <BillTo>
+                    <Name>
+                       <![CDATA[${order.billing_details.address.toName!}]]>
+                    </Name>
+                    <#--
+                    <Company>
+                       <![CDATA[XX]]>
+                    </Company>
+                    -->
+                    <Phone>
+                       <![CDATA[${order.billing_details.phone.contactNumber!}]]>
+                    </Phone>
+                    <Email>
+                       <![CDATA[${order.customer_details.email!}]]>
+                    </Email>
+                 </BillTo>
+                 <ShipTo>
+                    <#if !shipAddress.isEmpty()>
+                        <#assign geoState = ec.entity.find("moqui.basic.Geo").condition("geoId",shipAddress.stateProvinceGeoId!).useCache(true).one()>
+                        <#assign geoCountry = ec.entity.find("moqui.basic.Geo").condition("geoId",shipAddress.countryGeoId!).useCache(true).one()>
+                    </#if>
+                    <Name>
+                       <![CDATA[${shipAddress.toName!}]]>
+                    </Name>
+                    <#--
+                    <Company>
+                       <![CDATA[XX]]>
+                    </Company>
+                    -->
+                    <Address1>
+                       <![CDATA[${shipAddress.address1!}]]>
+                    </Address1>
+                    <Address2>${shipAddress.address2!}</Address2>
+                    <City>
+                       <![CDATA[${shipAddress.city!}]]>
+                    </City>
+                    <#if geoState??>
```

Comment:
What does "geoState??" condition means here?

Reply:
This is null check for geoState variable.

Reply:
This greatly reduced verbosity, thanks Siddharth!!